### ISSUE TITLE
phemex /api-data/spots/pnls doesnt return pnl?

### ISSUE THREAD

**Issue by jtashiro** [ccxt/ccxt#20571]
### Operating System

ubuntu and mac

### Programming Languages

Python

### CCXT Version

4.1.98

### Description

I'm calling  print(exchange.private_get_api_data_spots_pnls()) with verbose=True yields no data.   I have many spot buy/sell transactions and phemex website/app show pnl for several days.   

any ideas why this API call returns 0 rows/data?  Do I need to provide start/end and if so, how do I do that?

output:

CCXT Version: 4.1.98

fetch Request: phemex GET https://api.phemex.com/api-data/spots/pnls RequestHeaders: {'x-phemex-access-token': '65e53xxxxxxxxxx', 'x-phemex-request-expiry': '1703698489', 'x-phemex-request-signature': 'xxxxxx', 'User-Agent': 'python-requests/2.31.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: None

fetch Response: phemex GET https://api.phemex.com/api-data/spots/pnls 200 ResponseHeaders: {'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'Date': 'Wed, 27 Dec 2023 17:33:49 GMT', 'x-phemex-request-tracing': 'a052ee9d-8daa-43b3-a844-8616ea84c877', 'X-RateLimit-Remaining': '65', 'X-RateLimit-Capacity': '100', 'Vary': 'Origin, Access-Control-Request-Method, Access-Control-Request-Headers', 'Expires': 'Thu, 01 Jan 1970 00:00:00 GMT', 'X-Content-Type-Options': 'nosniff', 'X-XSS-Protection': '1; mode=block', 'X-Frame-Options': 'DENY', 'Referrer-Policy': 'no-referrer', 'X-Cache': 'Miss from cloudfront', 'Via': '1.1 c267e55887d5d0922d95439137f937dc.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'EWR50-C1', 'X-Amz-Cf-Id': 'qAZF_gQCBlgp0MP4agmsEckjN8fCQM0jbt70awo0389t-KmugrZNyw=='} 

ResponseBody: {"code":0,"msg":"OK","data":{"rows":[]}}
{'code': '0', 'msg': 'OK', 'data': {'rows': []}}

### Code

```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import sys
from pprint import pprint
import logging

root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(root + '/python')

import ccxt  # noqa: E402

import configparser
import app_config as cfg
from app_config import now, now_to_ms_epoch, datetime_to_ms_epoch


# -----------------------------------------------------------------------------

print('CCXT Version:', ccxt.__version__)

# -----------------------------------------------------------------------------

exchange = ccxt.phemex({
    'apiKey': cfg.key,
    'secret': cfg.secret,
    'verbose': True
})

try:
   print(exchange.private_get_api_data_spots_pnls())
except Exception as err:
    print(err)


```

**Comment by samgermain** [comment]
`private_get_api_data_spots_pnls` is an implicit exchange specific method, so we can't provide much more information than what the [phemex docs ](https://phemex-docs.github.io/#query-pnl) provide on this endpoint. 

- The api does mention that the parameter `start` defaults to 2 days before now, are you sure you've had PNL in the last 2 days? 
- Also was that PNL in your spot wallet, since this endpoint is specific to spot?

**Comment by jtashiro** [comment]
Yes to both - spot trading pnl daily for several weeks.  The phemex website and app show pnl history report online once/day 0 UTC and I’d like to see that more frequently ...

**Comment by carlotestor** [comment]
The empty `rows` is the endpoint's own default window, not a CCXT problem — `/api-data/spots/pnls` defaults to `start` = 2 days ago and returns only *settled daily* PnL rows. Since it's an implicit (exchange-specific) method, you pass the raw query params directly:

```python
import time
now = exchange.milliseconds()
resp = exchange.private_get_api_data_spots_pnls({
    'start': now - 30 * 24 * 60 * 60 * 1000,   # 30 days back, ms
    'end': now,
    'limit': 200,
    'offset': 0,
})
print(resp['data']['rows'])
```

Two caveats worth checking:
- The window is capped by Phemex (roughly 3 months); paginate with `offset` for longer ranges.
- The report is generated once a day at 00:00 UTC, so intraday PnL never appears here — that's why the app shows it but the API doesn't. For intra-day numbers you'd have to derive them from `fetch_my_trades()` yourself.

If passing explicit `start`/`end` returns your rows, this can be closed.
